### ISSUE TITLE
metadata in vcd_vapp doesn't work

### ISSUE THREAD

**Issue by nrobert13** [terraform-providers/terraform-provider-vcd#158]
Hi there,

### Terraform Version
Terraform v0.11.11

### Affected Resource(s)
Please list the resources as a list, for example:
- vcd_vapp

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl

resource "vcd_vapp" "node" {
  name = "vapp01"
  power_on = "false"
  metadata = {
    extradata = "extradata"
  }
  count = "2"
}


```

### Debug Output
* module.master.vcd_vapp.node[1]: 1 error(s) occurred:

* vcd_vapp.node.1: error adding metadata: &errors.errorString{s:"vApp doesn't contain any children, aborting customization"}
* module.master.vcd_vapp.node[0]: 1 error(s) occurred:

* vcd_vapp.node.0: error adding metadata: &errors.errorString{s:"vApp doesn't contain any children, aborting customization"}

### Expected Behavior
create vapp with metadata

### Actual Behavior
I'm not sure how is this metadata supposed to work in the vapp, because as far as I can see from the go-vcloud-director [code](https://github.com/vmware/go-vcloud-director/blob/7b3888d7e08e481b7aef9dc07c56cc5530ff33dc/govcd/vapp.go#L780), you cannot add metadata to an empty vapp, but as far as I understand the workflow in terraform is to add an empty vapp and then a vm into the vapp.

### Steps to Reproduce
1. `terraform apply`

**Comment by ybaumy** [comment]
Do not know exactly what you are trying to achieve. But you can specify an initscript which works with Windows VM's. If you want to pass metadata for cloud-init on Linux then you need the official vmware tools installed in the VM. open-vm-tools will not work. Since I wanted to use open-vm-tools the devs added the function to upload a ISO and attach it to the VM. With open-vm-tools and cloud-init only the OVF provider in cloud-init can be used. At least I havent found a way to do otherwise.
If you have further questions and the same problem as me then go and ask me ..

**Comment by nrobert13** [comment]
my problem is that the terraform-provider-vcd doesn't support to add metadata to the VM's. It used to work with v1.0.0, but it doesn't anymore.

**Comment by lvirbalas** [comment]
That's funky. We'll investigate. Thank you for the heads up.

**Comment by lvirbalas** [comment]
Related to issue #160

**Comment by Didainius** [comment]
This was resolved by @dn1s via #175 and is in master now. Target release - 2.2.0
